crates/resolver/src/lookup.rs

```diff
@@ -306,6 +333,33 @@ impl<'i> Iterator for SrvLookupIter<'i> {
     }
 }
 
+impl IntoIterator for SrvLookup {
+    type Item = rdata::SRV;
+    type IntoIter = SrvLookupIntoIter;
+
+    /// This is most likely not a free conversion, the RDatas will be cloned if data is
+    ///  held behind an Arc with more than one reference (which is most likely the case coming from cache)
+    fn into_iter(self) -> Self::IntoIter {
+        SrvLookupIntoIter(self.0.into_iter())
+    }
+}
+
+/// Borrowed view of set of RDatas returned from a Lookup
+pub struct SrvLookupIntoIter(LookupIntoIter);
+
+impl Iterator for SrvLookupIntoIter {
+    type Item = rdata::SRV;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let iter: &mut _ = &mut self.0;
```

Comment:
I'm surprised this type annotation is necessary?

Reply:
It was at one point, not sure if it still is...